### PR TITLE
NR-42250 DEBT: pay down TypeScript `any` usage webview

### PR DESCRIPTION
**Pull request by dsellarsnr**
- Use redux-toolkit to help with typing thunks
- Big step forward but does not close out use of `any`

### CODE REVIEW

File: vscode/src/api/tokenManager.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { Logger } from "logger";
 import { GlobalState } from "../common";
 import { extensionId } from "../constants";
 import { Container } from "../container";
-import { keychain } from "../keychain";
+import keychain from "keytar";
```
Comment (bcanzanella) [inline vscode/src/api/tokenManager.ts:6]: does this bypass that getNodeModule code?

Comment (bcanzanella) [reply vscode/src/api/tokenManager.ts:6]: ahh i see it in the externals... maybe a comment here to mention that?

File: shared/ui/store/streams/reducer.ts [hunk 2]
```diff
@@ -64,8 +64,8 @@ export const getChannelStreamsForTeam = createSelector(
 	(state: CodeStreamState) => state.streams,
 	(_, teamId: string) => teamId,
 	(state: CodeStreamState) => state.session.userId!,
-	(state, teamId, userId) => {
-		const streams = state.byTeam[teamId];
+	(streamState, teamId, userId) => {
+		const streams: CSStream[] = streamState.byTeam[teamId]; // TODO why is this not auto-typed
```
Comment (bcanzanella) [inline shared/ui/store/streams/reducer.ts:68]: hmm, maybe that empty `{}` on line 13?

File: vscode/src/api/tokenManager.ts (hunk 1 above)
Comment (dsellarsnr) [reply vscode/src/api/tokenManager.ts:6]: Yes, it was the cause of the scary looking eval warning in esbuild. Will add comment.

File: shared/ui/Stream/OpenPullRequests.tsx [hunk 3]
```diff
@@ -262,8 +262,8 @@ export const OpenPullRequests = React.memo((props: Props) => {
 			context.currentPullRequest.id;
 		const currentPullRequest = getCurrentProviderPullRequest(state);
 		const expandedPullRequestGroupIndex = context.currentPullRequest?.groupIndex;
-		const panePreferences = preferences.sidebarPanes || EMPTY_HASH;
-		const settings = panePreferences["open-pull-requests"] || EMPTY_HASH;
+		const panePreferences = preferences.sidebarPanes;
```
Comment (bcanzanella) [inline shared/ui/Stream/OpenPullRequests.tsx:265]: we ok not having these `EMPTY_HASH*`es?

File: shared/ui/Stream/PullRequest.tsx [hunk 4]
```diff
@@ -245,18 +241,14 @@ export const PullRequest = () => {
...
-			getPullRequestConversations(
-				derivedState.currentPullRequestProviderId!,
-				derivedState.currentPullRequestId!
-			)
-		)) as {
-			error?: {
-				message: string;
-			};
-		};
+		const response = await dispatch(
```
Comment (bcanzanella) [inline shared/ui/Stream/PullRequest.tsx:244]: nothing to do here. just thinking out loud. would there be any benefit to having a `typedDispatch` (or some other name) that runs an action and calls unwrap() on its own? is there overhead to calling unwrap()?

File: shared/ui/Stream/PullRequestFilesChangedFileComments.tsx [hunk 5]
```diff
@@ -136,7 +133,7 @@ export const PullRequestFilesChangedFileComments = (props: Props) => {
 
 	const syncCheckedStatusWithPr = () => {
 		if (currentPr && !isGitLab && supportsViewerViewedState) {
-			const prFiles = currentPr.files.nodes;
+			const prFiles = (currentPr as any).files.nodes; // TODO Fix typing - does this crash on gitlab since it files is only on github?
```
Comment (bcanzanella) [inline shared/ui/Stream/PullRequestFilesChangedFileComments.tsx:136]: i think the !isGitLab above guards that

File: shared/ui/Stream/OpenPullRequests.tsx (hunk 3 above)
Comment (dsellarsnr) [reply shared/ui/Stream/OpenPullRequests.tsx:265]: Ahh good catch.  I was getting rid of using {} as a type which is a eslint warning but it looks like it is needed here. Just not a fan of EMPTY_HASH - is {} ok?

File: shared/ui/Stream/PullRequest.tsx (hunk 4 above)
Comment (dsellarsnr) [reply shared/ui/Stream/PullRequest.tsx:244]: Will look into it

File: shared/ui/store/streams/reducer.ts (hunk 2 above)
Comment (dsellarsnr) [reply shared/ui/store/streams/reducer.ts:68]: The whole thing isn't typed - streamState is `any`. It should infer it from the selector return type. Managed to manually type it.

**Review by bcanzanella — APPROVED** [review]
🚀

**Pull request closed without merging**